core/src/main/scala/org/apache/spark/sql/execution/ExistingPlans.scala

```diff
@@ -104,8 +110,21 @@ private[sql] object PartitionedPhysicalScan {
       otherRDDs: Seq[RDD[InternalRow]],
       relation: PartitionedDataSourceScan,
       allFilters: Seq[Expression],
-      schemaAttributes: Seq[AttributeReference]): PartitionedPhysicalScan =
+      schemaAttributes: Seq[AttributeReference],
+      scanBuilderArgs: => (Seq[AttributeReference], Seq[Filter]) ): SparkPlan =
     relation match {
+      case i: IndexColumnFormatRelation =>
+        val columnScan = ColumnTableScan(output, rdd, otherRDDs, numBuckets,
+          partitionColumns, relation, allFilters, schemaAttributes)
+        val table = i.getBaseTableRelation
+        val (a, f) = scanBuilderArgs
+        val baseTableRDD = table.buildRowBufferRDD(Array.empty,
+          a.map(_.name).toArray, f.toArray, false)
+        val rowBufferScan = RowTableScan(output, baseTableRDD, numBuckets,
+          Seq.empty, table)
+        val bufferExchange = ShuffleExchange(columnScan.outputPartitioning,
+          rowBufferScan)
+        ZipPartitionScan(columnScan, bufferExchange)
```

Comment:
@ymahajan yeah :) atleast pushing the filters and predicates down to rowBuffers although non-pushable filters won't get in there :(.. Earlier it was pushing down filters that are non-pushable at ColumnBuffer level but now due to the way code warrants, i made it optional. Sumedh and me think that rowbuffer will be minimal and probably it is okay. If performance profiling shows otherwise we might need to push non-pushable fitlers and projection down there.